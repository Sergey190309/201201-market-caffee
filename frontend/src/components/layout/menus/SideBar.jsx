import React, { useState } from "react";
import { Link } from "react-router-dom";

import {
  Icon,
  Grid,
  Sidebar,
  Segment,
  Menu,
  Checkbox,
  Button,
} from "semantic-ui-react";

import Landing from "../../layout/Landing";
import Language from "../../language/Language";

const SideBar = () => {
  const [visible, setVisible] = useState(false);
  return (
    <Grid columns={1}>
      <Grid.Column>
        <Checkbox
          checked={visible}
          label={{ children: <code>visible</code> }}
          onChange={(e, data) => setVisible(data.checked)}
        />
      </Grid.Column>

      <Grid.Column>
        <Sidebar.Pushable as={Segment}>
          <Sidebar
            as={Menu}
            animation="overlay"
            icon="labeled"
            inverted
            onHide={() => setVisible(false)}
            vertical
            visible={visible}
            width="thin"
          >
            <Menu.Item name="logo">
              <Link to="/" onClick={() => setVisible(false)}>
                <Icon name="utensils" size="large" />
              </Link>
            </Menu.Item>
            <Menu.Item name="item01">NavBarItem01</Menu.Item>
            <Menu.Item name="item02">NavBarItem02</Menu.Item>
            <Menu.Item name="item03">NavBarItem03</Menu.Item>
            <Menu.Item name="login">
              <Link to="/login" onClick={() => setVisible(false)}>
                <Button>Log-In</Button>
              </Link>
            </Menu.Item>
            <Menu.Item name="language">
              <Link to="#">
                <Language />
              </Link>
            </Menu.Item>
          </Sidebar>
          <Sidebar.Pusher dimmed={visible}>
            <Segment basic>
              <Landing />
            </Segment>
          </Sidebar.Pusher>
        </Sidebar.Pushable>
      </Grid.Column>
    </Grid>
  );
};

export default SideBar;
