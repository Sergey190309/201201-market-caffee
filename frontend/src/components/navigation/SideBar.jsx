import React, { useState } from "react";
import { Link } from "react-router-dom";

import { Icon, Grid, Sidebar, Segment, Menu, Button } from "semantic-ui-react";

import Content from "../content/Content";
import Logo from '../nav_items/Logo'
// import Language from "../../language/Language";

const SideBar = () => {
  const [visible, setVisible] = useState(false);

  return (
    <Grid celled>
      <Grid.Row color='teal'>
        <Grid.Column width="2" textAlign='center' verticalAlign='middle'>
          <Logo />
        </Grid.Column>
        <Grid.Column width="12" textAlign='center' verticalAlign='middle'>
          <h1>Hi there!</h1>
        </Grid.Column>
        <Grid.Column width="2" textAlign='center' verticalAlign='middle'>
          <Icon
            name={visible ? "angle double right" : "angle double left"}
            size="big"
            onClick={() => setVisible(!visible)}
          />
        </Grid.Column>
      </Grid.Row>

      <Grid.Row>
        <Grid.Column>
          <Sidebar.Pushable as={Segment}>
            <Sidebar
              as={Menu}
              animation="overlay"
              icon="labeled"
              inverted
              onHide={() => setVisible(false)}
              vertical
              direction="right"
              visible={visible}
              width="thin"
            >
              <Menu.Item name="item01">NavBarItem01</Menu.Item>
              <Menu.Item name="item02">NavBarItem02</Menu.Item>
              <Menu.Item name="item03">NavBarItem03</Menu.Item>
              <Menu.Item name="login">
                <Link to="/login" onClick={() => setVisible(false)}>
                  <Button>Log-In</Button>
                </Link>
              </Menu.Item>
              {/* <Menu.Item name="language">
                <Link to="#">
                  <Language />
                </Link>
              </Menu.Item> */}
            </Sidebar>
            <Sidebar.Pusher dimmed={visible}>
              <Segment basic>
                <Content />
              </Segment>
            </Sidebar.Pusher>
          </Sidebar.Pushable>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
};

export default SideBar;
