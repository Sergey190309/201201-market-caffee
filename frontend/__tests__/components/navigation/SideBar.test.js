import React from "react";
// import { render } from "@testing-library/react";
import ShallowRenderer from "react-test-renderer/shallow";

import SideBar from '../../../src/components/navigation/SideBar'

describe("Layout compnent", () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<SideBar />)
  test("Snapshot", () => {
    expect(result).toMatchSnapshot()
  });
});
