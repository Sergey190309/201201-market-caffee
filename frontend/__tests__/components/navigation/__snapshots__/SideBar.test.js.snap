// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout compnent Snapshot 1`] = `
<Grid
  celled={true}
>
  <GridRow
    color="teal"
  >
    <GridColumn
      textAlign="center"
      verticalAlign="middle"
      width="2"
    >
      <Logo />
    </GridColumn>
    <GridColumn
      textAlign="center"
      verticalAlign="middle"
      width="12"
    >
      <h1>
        Hi there!
      </h1>
    </GridColumn>
    <GridColumn
      textAlign="center"
      verticalAlign="middle"
      width="2"
    >
      <Icon
        as="i"
        name="angle double left"
        onClick={[Function]}
        size="big"
      />
    </GridColumn>
  </GridRow>
  <GridRow>
    <GridColumn>
      <SidebarPushable
        as={[Function]}
      >
        <Sidebar
          animation="overlay"
          as={[Function]}
          direction="right"
          icon="labeled"
          inverted={true}
          onHide={[Function]}
          target={
            Object {
              "current": Document {
                "location": Location {
                  "assign": [Function],
                  "hash": "",
                  "host": "localhost",
                  "hostname": "localhost",
                  "href": "http://localhost/",
                  "origin": "http://localhost",
                  "pathname": "/",
                  "port": "",
                  "protocol": "http:",
                  "reload": [Function],
                  "replace": [Function],
                  "search": "",
                  "toString": [Function],
                },
                Symbol(DOM SymbolTree): SymbolTreeNode {
                  "cachedIndex": -1,
                  "cachedIndexVersion": NaN,
                  "childIndexCachedUpTo": null,
                  "childrenVersion": 0,
                  "firstChild": null,
                  "lastChild": null,
                  "nextSibling": null,
                  "parent": null,
                  "previousSibling": null,
                },
              },
            }
          }
          vertical={true}
          visible={false}
          width="thin"
        >
          <MenuItem
            name="item01"
          >
            NavBarItem01
          </MenuItem>
          <MenuItem
            name="item02"
          >
            NavBarItem02
          </MenuItem>
          <MenuItem
            name="item03"
          >
            NavBarItem03
          </MenuItem>
          <MenuItem
            name="login"
          >
            <Link
              onClick={[Function]}
              to="/login"
            >
              <Button
                as="button"
              >
                Log-In
              </Button>
            </Link>
          </MenuItem>
        </Sidebar>
        <SidebarPusher
          dimmed={false}
        >
          <Segment
            basic={true}
          >
            <Content />
          </Segment>
        </SidebarPusher>
      </SidebarPushable>
    </GridColumn>
  </GridRow>
</Grid>
`;
